README.md
# Bgoonz Blog Readme

<div align="centered">

### [⇨WEBSITE🗺️⇦](https://bgoonz-blog.netlify.app)

### [CHANGELOG](./CHANGELOG.md)

##### [⇨WEBSITE🗺️⇦](https://bgoonz-blog.netlify.app) [⇨**Cloudfare-Backup**⇦](https://bgoonz-blog-2-0.pages.dev/) [⇨**search**⇦](https://www.algolia.com/realtime-search-demo/web-dev-resource-hub-9e6b8aa8-6106-44c5-9f59-ff3f9531abd4) [⇨**Backup Repo Deploy**⇦](https://bgoonzblog20-backup.netlify.app/#gsc.tab=0) [⇨**Github pages**⇦](https://bgoonz.github.io/BGOONZ_BLOG_2.0/) [⇨**Go To Site Wiki**⇦](https://github.com/Backup242/BGOONZ_BLOG_2.0/wiki) [⇨**Gatsby Cloud Version**⇦](https://bgoonzblog20master.gatsbyjs.io/) [⇨**Vercel Version**⇦](https://bgoonz-blog-2-0.vercel.app/) [⇨**Cloudfare-Domain**⇦](www.webdevhub.us) [⇨**gh-pages**⇦](https://bgoonz.github.io/BGOONZ_BLOG_2.0/) [⇨**backup netlify deploy**⇦](https://bgoonz-blog20-backup.netlify.app/)

### Repos

[Alt Backup](https://github.com/BGOOONZ-BLOG/bgoonz-blog2.0-v-5)
[Blog Content](https://github.com/bgoonz/MainBlogContent)

#### [⇨Privacy policy⇦](https://codepen.io/bgoonz/pen/LYLJZrW)

[![Netlify Status](https://api.netlify.com/api/v1/badges/a1b7ee1a-11a7-4bd2-a341-2260656e216f/deploy-status)](https://app.netlify.com/sites/bgoonz-blog/deploys)[![CodeFactor](https://www.codefactor.io/repository/github/bgoonz/bgoonz_blog_2.0/badge)](https://www.codefactor.io/repository/github/bgoonz/bgoonz_blog_2.0)

![GitHub visitors](https://visitor-badge-reloaded.herokuapp.com/badge?page_id=bgoonz.visitor.badge.reloaded&color=00bbbb&style=for-the-badge&logo=github)

[![Demo](https://img.youtube.com/vi/OGCcq1_Tbzk/0.jpg)](https://www.youtube.com/watch?v=OGCcq1_Tbzk)

</div>
---

<!-- @import "[TOC]" {cmd="toc" depthFrom=1 depthTo=6 orderedList=false} -->

<!-- code_chunk_output -->

- [Bgoonz Blog Readme](#bgoonz-blog-readme)
  - [⇨WEBSITE🗺️⇦](#website️httpsbgoonz-blognetlifyapp)
  - [CHANGELOG](#changelogchangelogmd)
    - [⇨WEBSITE🗺️⇦ ⇨**Cloudfare-Backup**⇦ ⇨**search**⇦ ⇨**Backup Repo Deploy**⇦ ⇨**Github pages**⇦ ⇨**Go To Site Wiki**⇦ ⇨**Gatsby Cloud Version**⇦ ⇨**Vercel Version**⇦ ⇨**Cloudfare-Domain**⇦ ⇨**gh-pages**⇦ ⇨**backup netlify deploy**⇦](#website️httpsbgoonz-blognetlifyapp-cloudfare-backuphttpsbgoonz-blog-2-0pagesdev-searchhttpswwwalgoliacomrealtime-search-demoweb-dev-resource-hub-9e6b8aa8-6106-44c5-9f59-ff3f9531abd4-backup-repo-deployhttpsbgoonzblog20-backupnetlifyappgsctab0-github-pageshttpsbgoonzgithubiobgoonz_blog_20-go-to-site-wikihttpsgithubcombgoonzbgoonz_blog_20wiki-gatsby-cloud-versionhttpsbgoonzblog20mastergatsbyjsio-vercel-versionhttpsbgoonz-blog-2-0vercelapp-cloudfare-domainwwwwebdevhubus-gh-pageshttpsbgoonzgithubiobgoonz_blog_20-backup-netlify-deployhttpsbgoonz-blog20-backupnetlifyapp)
  - [Repos](#repos)
    - [⇨Privacy policy⇦](#privacy-policyhttpscodepeniobgoonzpenlyljzrw)
  - [Useful Commands (internal use)](#useful-commands-internal-use)
  - [Web Dev Setup](#web-dev-setup)
  - [Bash Cheat Sheet (MY COMMANDS)](#bash-cheat-sheet-my-commands)
    - [Find](#find)
  - [To find files by case-insensitive extension (ex: .jpg, .JPG, .jpG)](#to-find-files-by-case-insensitive-extension-ex-jpg-jpg-jpg)
  - [To find directories](#to-find-directories)
  - [To find files](#to-find-files)
  - [To find files by octal permission](#to-find-files-by-octal-permission)
  - [To find files with setuid bit set](#to-find-files-with-setuid-bit-set)
  - [To find files with extension '.txt' and remove them](#to-find-files-with-extension-txt-and-remove-them)
  - [To find files with extension '.txt' and look for a string into them](#to-find-files-with-extension-txt-and-look-for-a-string-into-them)
  - [To find files with size bigger than 5 Mebibyte and sort them by size](#to-find-files-with-size-bigger-than-5-mebibyte-and-sort-them-by-size)
  - [To find files bigger than 2 Megabyte and list them](#to-find-files-bigger-than-2-megabyte-and-list-them)
  - [To find files modified more than 7 days ago and list file information](#to-find-files-modified-more-than-7-days-ago-and-list-file-information)
  - [To find symlinks owned by a user and list file information](#to-find-symlinks-owned-by-a-user-and-list-file-information)
  - [To search for and delete empty directories](#to-search-for-and-delete-empty-directories)
  - [To search for directories named build at a max depth of 2 directories](#to-search-for-directories-named-build-at-a-max-depth-of-2-directories)
  - [To search all files who are not in .git directory](#to-search-all-files-who-are-not-in-git-directory)
  - [To find all files that have the same node (hard link) as MY_FILE_HERE](#to-find-all-files-that-have-the-same-node-hard-link-as-my_file_here)
  - [To find all files in the current directory and modify their permissions](#to-find-all-files-in-the-current-directory-and-modify-their-permissions)
  - [Wiki Nav](#wiki-nav)
  - [Dependencies](#dependencies)
  - [Docs Structure](#docs-structure)
    - [Overview](#overview)
    - [Syntax](#syntax)
    - [Parameters](#parameters)
    - [Return value](#return-value)
    - [Example](#example)
    - [Explanation](#explanation)
    - [Output](#output)
  - [Technoloy](#technoloy)
  - [SOURCECODE](#sourcecode)
    - [Overview](#overview-1)
    - [Syntax](#syntax-1)
    - [Parameters](#parameters-1)
    - [Return value](#return-value-1)
    - [Example](#example-1)
    - [Explanation](#explanation-1)
    - [Output](#output-1)
  - [Source Code](#source-code)
  - [Component Structure](#component-structure)

<!-- /code_chunk_output -->

![preview](https://github.com/Backup242/BGOONZ_BLOG_2.0/blob/master/static/images/blog-may-2022.png?raw=true)

---

## Useful Commands (internal use)

```bash

npm i @stackbit/gatsby-plugin-menus babel-runtime chokidar classnames fs-extra gatsby gatsby-image gatsby-plugin-netlify gatsby-plugin-netlify-cms gatsby-plugin-sass gatsby-plugin-react-helmet gatsby-plugin-typescript  --legacy-peer-deps


git checkout 307a5cd        # check out the commit that you want to reset to
git checkout -b fixy        # create a branch named fixy to do the work
git merge -s ours master    # merge master's history without changing any files
git checkout master         # switch back to master
git merge fixy              # and merge in the fixed branch
git push                    # done, no need to force push!



```

---

<a href="https://testmysite.io/61e5ff836a840eeeef7e78e9/bgoonz-blog.netlify.app" rel="Test My Site Results">
![Test My Site](https://github.com/Backup242/BGOONZ_BLOG_2.0/blob/master/static/images/Screenshot%202022-03-30%20at%2012-37-24%20Website%20Speed%20Test%20Tool%20-%20Testmysite.io%20by%20Netlify.png?raw=true)</a>

---

## Web Dev Setup

<details>

<summary>Web Dev Setup</summary>

---

title: WebDev Setup
template: post
subtitle: Basic Web Development Environment Setup
excerpt: Windows Subsystem for Linux (WSL) and Ubuntu
date: 2022-05-08T18:10:03.296Z
image: <https://cdn-images-1.medium.com/max/800/0>*aqKP1drNHmNm34zz.jpg
thumb_image: <https://cdn-images-1.medium.com/max/800/0>*aqKP1drNHmNm34zz.jpg
image_position: right
author: src/data/authors/bgoonz.yaml
categories: - src/data/categories/js.yaml - src/data/categories/javascript.yaml
tags: - src/data/tags/links.yaml - src/data/tags/resources.yaml
show_author_bio: true
related_posts: - src/pages/blog/vs-code-extensions.md - src/pages/blog/code-playgrounds-of-2021.md - src/pages/blog/adding-css-to-your-html.md - src/pages/blog/deploy-react-app-to-heroku.md
cmseditable: true

---

## Basic Web Development Environment Setup

Windows Subsystem for Linux (WSL) and Ubuntu

---

### Basic Web Development Environment Setup

#### Windows Subsystem for Linux (WSL) and Ubuntu

<figure>
<img src="https://cdn-images-1.medium.com/max/800/0*aqKP1drNHmNm34zz.jpg" class="graf-image" />
</figure>Test if you have Ubuntu installed by typing "Ubuntu" in the search box in the bottom app bar that reads "Type here to search". If you see a search result that reads **"Ubuntu 20.04 LTS"** with "App" under it, then you have it installed.
```
1.  In the application search box in the bottom bar, type "PowerShell" to find the application named "Windows PowerShell"
2.  Right-click on "Windows PowerShell" and choose "Run as administrator" from the popup menu
3.  In the blue PowerShell window, type the following: `Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux`
4.  Restart your computer
5.  In the application search box in the bottom bar, type "Store" to find the application named "Microsoft Store"
6.  Click "Microsoft Store"
7.  Click the "Search" button in the upper-right corner of the window
8.  Type in "Ubuntu"
9.  Click "Run Linux on Windows (Get the apps)"
10. Click the orange tile labeled **"Ubuntu"** Note that there are 3 versions in the Microsoft Store… you want the one just entitled 'Ubuntu'
11. Click "Install"
12. After it downloads, click "Launch"
13. If you get the option, pin the application to the task bar. Otherwise, right-click on the orange Ubuntu icon in the task bar and choose "Pin to taskbar"
14. When prompted to "Enter new UNIX username", type your first name with no spaces
15. When prompted, enter and retype a password for this UNIX user (it can be the same as your Windows password)
16. Confirm your installation by typing the command `whoami 'as in who-am-i'`followed by Enter at the prompt (it should print your first name)
17. You need to update your packages, so type `sudo apt update` (if prompted for your password, enter it)
18. You need to upgrade your packages, so type `sudo apt upgrade` (if prompted for your password, enter it)
```
### Git

Git comes with Ubuntu, so there's nothing to install. However, you should configure it using the following instructions.

Open an Ubuntu terminal if you don't have one open already.

1. > You need to configure Git, so type `git config --global user.name "Your Name"` with replacing "Your Name" with your real name.
2. > You need to configure Git, so type `git config --global user.email your@email.com` with replacing "<a href="mailto:your@email.com" class="markup--anchor markup--li-anchor">your@email.com</a>" with your real email.

**Note: if you want git to remember your login credentials type:**

    git config --global credential.helper store

### Google Chrome

Test if you have Chrome installed by typing "Chrome" in the search box in the bottom app bar that reads "Type here to search". If you see a search result that reads "Chrome" with "App" under it, then you have it installed. Otherwise, follow these instructions to install Google Chrome.

### Node.js

Test if you have Node.js installed by opening an Ubuntu terminal and typing `node --version`. If it reports "Command 'node' not found", then you need to follow these directions.

1. > In the Ubuntu terminal, type `sudo apt update` and press Enter
2. > In the Ubuntu terminal, type `sudo apt install build-essential` and press Enter
3. > In the Ubuntu terminal, type `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | bash` and press Enter
4. > In the Ubuntu terminal, type `. ./.bashrc` and press Enter
5. > In the Ubuntu terminal, type `nvm install --lts` and press Enter
6. > Confirm that **node** is installed by typing `node --version` and seeing it print something that is not "Command not found"!

### Unzip

You will often have to download a zip file and unzip it. It is easier to do this from the command line. So we need to install a linux unzip utility.

In the Ubuntu terminal type: `sudo apt install unzip` and press Enter

Mocha.js

Test if you have Mocha.js installed by opening an Ubuntu terminal and typing `which mocha`. If it prints a path, then you're good. Otherwise, if it prints nothing, install Mocha.js by typing `npm install -g mocha`.

### Python 3

Ubuntu does not come with Python 3. Install it using the command `sudo apt install python3`. Test it by typing `python3 --version` and seeing it print a number.

### Note about WSL

As of the time of writing of this document, WSL has an issue renaming or deleting files if Visual Studio Code is open. So before doing any linux commands which manipulate files, make sure you **close** Visual Studio Code before running those commands in the Ubuntu terminal.

### Some other common instillations

```bash
    # Installing build essentials
    sudo apt-get install -y build-essential libssl-dev
    # Nodejs and NVM
    curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh | bash
    source ~/.profile
    sudo nvm install 7.10.0
    sudo nvm use 7.10.0
    node -v
    #nodemon
    sudo npm install -g nodemon
    sudo npm install -g loopback-cli
    # Forever to run nodejs scripts forever
    sudo npm install forever -g
    # Git - a version control system
    sudo apt-get update
    sudo apt-get install -y git xclip
    # Grunt - an automated task runner
    sudo npm install -g grunt-cli
    # Bower - a dependency manager
    sudo npm install -g bower
    # Yeoman - for generators
    sudo npm install -g yo
    # maven
    sudo apt-get install maven -y
    # Gulp - an automated task runner
    sudo npm install -g gulp-cli
    # Angular FullStack - My favorite MEAN boilerplate (MEAN = MongoDB, Express, Angularjs, Nodejs)
    sudo npm install -g generator-angular-fullstack
    # Vim, Curl, Python - Some random useful stuff
    sudo apt-get install -y vim curl python-software-properties
    sudo apt-get install -y python-dev, python-pip
    sudo apt-get install -y libkrb5-dev
    # Installing JDK and JRE
    sudo apt-get install -y default-jre
    sudo apt-get install -y default-jdk
    # Archive Extractors
    sudo apt-get install -y unace unrar zip unzip p7zip-full p7zip-rar sharutils rar uudeview mpack arj cabextract file-roller
    # FileZilla - a FTP client
    sudo apt-get install -y filezilla
```

</details>

---

### Documentation Commands

documentation build src/components/DocsMenu.js  -f md> ./notes/DocsMenu.md
documentation build src/components/Footer.js  -f md> ./notes/Footer.md
documentation build src/components/Header.js  -f md> ./notes/Header.md
documentation build src/components/index.js  -f md> ./notes/index.md
documentation build src/components/SectionContent.js  -f md> ./notes/SectionContent.md
documentation build src/components/SectionDocs.js  -f md> ./notes/SectionDocs.md
documentation build src/components/SectionHero.js  -f md> ./notes/SectionHero.md
documentation build src/components/CtaButtons.js  -f md> ./notes/CtaButtons.md
documentation build src/components/DocsSubmenu.js  -f md> ./notes/DocsSubmenu.md
documentation build src/components/FormField.js  -f md> ./notes/FormField.md
documentation build src/components/Icon.js  -f md> ./notes/Icon.md
documentation build src/components/Layout.js  -f md> ./notes/Layout.md
documentation build src/components/SectionCta.js  -f md> ./notes/SectionCta.md
documentation build src/components/SectionGrid.js  -f md> ./notes/SectionGrid.md
documentation build src/components/Submenu.js  -f md> ./notes/Submenu.md
documentation build src/components/ActionLink.js  -f md> ./notes/ActionLink.md

---

## Blog Content

---

---

## Bash Cheat Sheet (MY COMMANDS)

<details>

<summary> Bash Cheat Sheet (MY COMMANDS) </summary

## 1. Remove spaces from file and folder names and then remove numbers from files and folder names

### Description: need to : `sudo apt install rename`

> Notes: Issue when renaming file without numbers collides with existing file name...

###### code

```bash
find . -name "* *" -type d | rename 's/ /_/g'
find . -name "* *" -type f | rename 's/ /_/g'
```

```bash

```bash
find $dir -type f | sed 's|\(.*/\)[^A-Z]*\([A-Z].*\)|mv \"&\" \"\1\2\"|' | sh

find $dir -type d | sed 's|\(.*/\)[^A-Z]*\([A-Z].*\)|mv \"&\" \"\1\2\"|' | sh

for i in *.html; do mv "$i" "${i%-*}.html"; done

for i in *.*; do mv "$i" "${i%-*}.${i##*.}"; done

---
### Description: combine the contents of every file in the contaning directory.

>Notes: this includes the contents of the file it's self...

###### code:

```js
//APPEND-DIR.js
const fs = require('fs');
let cat = require('child_process')
  .execSync('cat *')
  .toString('UTF-8');
fs.writeFile('output.md', cat, err => {
  if (err) throw err;
});

```

---

## 2. Download Website Using Wget

### Description

> Notes: ==> sudo apt install wget

###### code

```bash

wget --limit-rate=200k --no-clobber --convert-links --random-wait -r -p -E -e robots=off -U mozilla https://bootcamp42.gitbook.io/python/

```

---

## 3. Clean Out Messy Git Repo

### Description: recursively removes git related folders as well as internal use files / attributions in addition to empty folders

> Notes: To clear up clutter in repositories that only get used on your local machine.

###### code

````bash

find . -empty -type d -print -delete

find . \( -name ".git" -o -name ".gitignore" -o -name ".gitmodules" -o -name ".gitattributes" \) -exec rm -rf -- {} +

find . \( -name "*SECURITY.txt" -o -name "*RELEASE.txt" -o  -name "*CHANGELOG.txt" -o -name "*LICENSE.txt" -o -name "*CONTRIBUTING.txt" -name "*HISTORY.md" -o -name "*LICENSE" -o -name "*SECURITY.md" -o -name "*RELEASE.md" -o  -name "*CHANGELOG.md" -o -name "*LICENSE.md" -o -name "*CODE_OF_CONDUCT.md" -o -name "*CONTRIBUTING.md" \) -exec rm -rf -- {} +

```

---

## 4. clone all of a user's git repositories

### Description: clone all of a user or organization's git repositories

> Notes:

###### code

## Generalized

```bash

CNTX={users|orgs}; NAME={username|orgname}; PAGE=1
curl "https://api.github.com/$CNTX/$NAME/repos?page=$PAGE&per_page=100" |
  grep -e 'git_url*' |
  cut -d \" -f 4 |
  xargs -L1 git clone
```

## Clone all Git User

```bash
CNTX={users}; NAME={bgoonz}; PAGE=1
curl "https://api.github.com/$CNTX/$NAME/repos?page=$PAGE&per_page=200"?branch=master |
  grep -e 'git_url*' |
  cut -d \" -f 4 |
  xargs -L1 git clone

```

## Clone all Git Organization

```bash

CNTX={organizations}; NAME={TheAlgorithms}; PAGE=1
curl "https://api.github.com/$CNTX/$NAME/repos?page=$PAGE&per_page=200"?branch=master |
  grep -e 'git_url*' |
  cut -d \" -f 4 |
  xargs -L1 git clone

```

---

## 5. Git Workflow

### Description

###### code

```bash
git pull
git init
git add .
git commit -m"update"
git push -u origin master
```

```bash
git init
git add .
git commit -m"update"
git push -u origin main
```

```bash

git init
git add .
git commit -m"update"
git push -u origin bryan-guner
```

```bash
git init
git add .
git commit -m"update"
git push -u origin gh-pages
````

```bash
git init
git add .
git commit -m"update"
git push -u origin preview
```

---

## 6. Recursive Unzip In Place

### Description: recursively unzips folders and then deletes the zip file by the same name

> Notes:

###### code

```bash

find . -name "*.zip" | while read filename; do unzip -o -d "`dirname "$filename"`" "$filename"; done;

find . -name "*.zip" -type f -print -delete

```

---

## 7. git pull keeping local changes

### Description

> Notes:

###### code

```bash

git stash
git pull
git stash pop

```

---

## 8. Prettier Code Formatter

### Description

> Notes:

###### code

```bash

sudo npm i prettier -g

prettier --write .

```

---

## 9. Pandoc

### Description

> Notes:

###### code

```bash

find ./ -iname "*.md" -type f -exec sh -c 'pandoc --standalone "${0}" -o "${0%.md}.html"' {} \;


CHANGELOG.md
Git History on master by all authors

<details>

<summary>
 Original

</summary>


Merge pull request #1564 from bgoonz/master
3e3b8baab (HEAD -> master, origin/master, origin/HEAD) by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)

Merge pull request #1563 from bgoonz/preview
ec957a352 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)

arrow-functions.md: changed
11c02b629 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 485 insertions(+)

arrow-functions.md: changed
7810402c8 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 1 deletion(-)

arrow-functions.md: changed
88007d5b6 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 1 deletion(-)

arrow-functions.md: changed
c2c171e14 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 1 deletion(-)

arrow_functions.md: changed
c289b78b9 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 482 deletions(-)

arrow_functions.md: changed
c50b15f4c by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 33 insertions(+), 33 deletions(-)

arrow_functions.md: changed
53bfee217 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 482 insertions(+)

arrow_functions.md: changed
2ef2eeb91 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 482 deletions(-)

bigo.md: removed
b08a46211 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 99 deletions(-)

big-o.md: changed
e1c693189 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 7 insertions(+)

bigo.md: changed
74350efe3 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 7 deletions(-)

bigo.md: changed
0f1af8c67 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 1 deletion(-)

bigo.md: changed
34dabe28f by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 1 deletion(-)

bigo.md: changed
829b8e7fa by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 1 deletion(-)

\_animations.scss: removed
9b1cccc77 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 0 insertions(+), 0 deletions(-)

main.yml: changed
95d2e3cfe by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 0 insertions(+), 0 deletions(-)

Merge pull request #1562 from bgoonz/master
5da883428 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)

Create main.yml
4e1760df2 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 24 insertions(+)

Add files via upload
029dbf868 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)
16 files changed, 3468 insertions(+), 1968 deletions(-)

first-100.md: changed
2b899d31a by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 12 insertions(+), 1 deletion(-)

first-100.md: changed
c43e50b20 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 11042 insertions(+)

first-100.md: added
7be93fb72 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 0 insertions(+), 0 deletions(-)

Merge pull request #1561 from bgoonz/master
ffb04efe8 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)

Merge pull request #1560 from bgoonz/preview
68d7afa15 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)

site-metadata.json: Multiple updates: _updated social_links in footer_ updated links in footer \* updated new_window in links
cb64efe0c by Bryan Guner <bryan.guner@gmail.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 17 insertions(+), 1 deletion(-)

site-metadata.json: updated style in links
02e19f657 by Bryan Guner <bryan.guner@gmail.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 1 deletion(-)

site-metadata.json: updated url in links
681dc1538 by Bryan Guner <bryan.guner@gmail.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 1 deletion(-)

site-metadata.json: updated label in links
e71546bfb by Bryan Guner <bryan.guner@gmail.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 1 insertion(+), 2 deletions(-)

Merge remote-tracking branch 'origin/preview' into preview
d359c6d0c by Stackbit <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)

sandbox.config.json: removed
62c064142 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 8 deletions(-)

Merge pull request #1559 from bgoonz/master
f3d773a04 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)

Merge pull request #1558 from bgoonz/preview
11c956350 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)

sandbox.config.json: removed
891f7ce02 by Stackbit Code Editor <projects@stackbit.com>, Sat Jan 22 2022 (2 days ago)
1 file changed, 8 deletions(-)

Merge pull request #1557 from bgoonz/master
8d62008a2 by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)

Add files via upload
9c92457ca by Bryan C Guner <66654881+bgoonz@users.noreply.github.com>, Sat Jan 22 2022 (2 days ago)
5 files changed, 249 insertions(+), 250 deletions(-)

Merge pull request #1556 from bgoonz/master

src/components/Header.js
/* It's a header component that renders a logo, a navbar, and a search bar. */
import _ from 'lodash';
import React from 'react';
import {classNames, Link, withPrefix} from '../utils';
import ActionLink from './ActionLink';
import Submenu from './Submenu';
export default class Header extends React.Component {
    render() {
        return (
            <header id="masthead" className="site-header outer">
                <br/>
                <div className="inner">
                    <div className="site-header-inside">
                        <div className="site-branding">
                            {
                            _.get(this.props, 'pageContext.site.siteMetadata.header.logo_img', null) ? (
                                <p className="site-logo">
                                    <Link to={
                                        withPrefix(_.get(this.props, 'pageContext.site.siteMetadata.header.url', null) || '/')
                                    }>
                                        <img src={
                                                withPrefix(_.get(this.props, 'pageContext.site.siteMetadata.header.logo_img', null))
                                            }
                                            alt={
                                                _.get(this.props, 'pageContext.site.siteMetadata.header.logo_img_alt', null)
                                            }/>
                                    </Link>
                                </p>
                            ) : (
                                <p className="site-logo">
                                    <Link to={
                                        withPrefix(_.get(this.props, 'pageContext.site.siteMetadata.header.url', null) || '/')
                                    }>
                                        {
                                        _.get(this.props, 'pageContext.site.siteMetadata.header.title', null)
                                    } </Link>
                                </p>
                            )
                        } </div>
                        {
                        _.get(this.props, 'pageContext.site.siteMetadata.header.has_nav', null) && (
                            <React.Fragment>
                                <nav id="main-navigation" className="site-navigation" aria-label="Main Navigation">
                                    <div className="site-nav-inside">
                                        <button id="menu-close" className="menu-toggle">
                                            <span className="screen-reader-text">Open Menu</span>
                                            <span className="icon-close" aria-hidden="true"/>
                                        </button>
                                        <ul className="menu">
                                            {
                                            _.map(_.get(this.props, 'pageContext.site.siteMetadata.header.nav_links', null), (action, action_idx) => {
                                                let page_url = _.trim(_.get(this.props, 'pageContext.url', null), '/');
                                                let action_url = _.trim(_.get(action, 'url', null), '/');
                                                return (
                                                    <li key={action_idx}
                                                        className={
                                                            classNames('menu-item', {
                                                                'has-children': _.get(action, 'has_subnav', null) && _.get(action, 'subnav_links', null),
                                                                current: page_url === action_url,
                                                                'menu-button': _.get(action, 'style', null) !== 'link'
                                                            })
                                                    }>
                                                        <ActionLink { ...this.props }
                                                            action={action}/> {
                                                        _.get(action, 'has_subnav', null) && _.get(action, 'subnav_links', null) && (
                                                            <React.Fragment>
                                                                <button className="submenu-toggle">
                                                                    <span className="icon-angle-right" aria-hidden="true"/>
                                                                    <span className="screen-reader-text">Sub-menu</span>
                                                                </button>
                                                                <Submenu { ...this.props }
                                                                    submenu={
                                                                        _.get(action, 'subnav_links', null)
                                                                    }
                                                                    menu_class={'submenu'}
                                                                    page={
                                                                        this.props.pageContext
                                                                    }/>
                                                            </React.Fragment>
                                                        )
                                                    } </li>
                                                );
                                            })
                                        } </ul>
                                    </div>
                                </nav>
                                <button id="menu-open" className="menu-toggle">
                                    <span className="screen-reader-text">Close Menu</span>
                                    <span className="icon-menu" aria-hidden="true"/>
                                </button>
                            </React.Fragment>
                        )
                    } </div>
                </div>
                <>
                    <div id="sb-search-example">
                        <div className="sb-search-icon" id="myCustomSearchButtonID">
                            <i className="sb-icon"></i>
                            <p>Search</p>
                        </div>
                        <style dangerouslySetInnerHTML={
                            {__html: "\n\t\t@font-face {\n            font-family: font-sb;\n            src: url(https://api.searchbar.org/fonts/sb.eot?39450784);\n            src: url(https://api.searchbar.org/fonts/sb.eot?39450784#iefix) format('embedded-opentype'), url(https://api.searchbar.org/fonts/sb.woff?39450784) format('woff'), url(https://api.searchbar.org/fonts/?39450784) format('truetype'), url(https://api.searchbar.org/fonts/sb.svg?39450784#font-sb) format('svg');\n            font-weight: 400;\n            font-style: normal\n        }\n\n        #sb-search-example p {\n            font-family: sans-serif;\n            font-size: 18px;\n            line-height: 12px;\n\t        margin: 0;\n        }\n\n        #sb-search-example {\n            position: static;\n            top: 0;\n            width: auto;\n            place-content: center;\n            display: inline-block;\n            box-shadow: 0px 2px 4px rgba(190, 190, 190, 0.5);\n            border: solid 5px white;\n            border-radius: 100px;\n            flex-shrink: initial;\n            flex-grow: initial;\n        }\n\n\n        #sb-search-example .sb-icon {\n            font: normal normal 400 26px font-sb;\n            align-items: center;\n            padding: 10px;\n            speak: none;\n            display: inline-block;\n            text-decoration: inherit;\n            text-align: center;\n            text-transform: none;\n            cursor: pointer;\n            margin: 0;\n        }\n\n        #sb-search-example .sb-search-icon {\n            box-sizing: border-box;\n            border: 0px;\n            align-items: center;\n            background: #007cff;\n            color: #fff;\n            cursor: pointer;\n            display: flex;\n            border-radius: 26px;\n            padding: 0 20px 0 4px;\n        }\n\t"}
                        }/>
                    </div>
                </>

                <a className="github-corner" href="https://github.com/Backup242/BGOONZ_BLOG_2.0" aria-label="View source on Github">
                    <svg aria-hidden="true"
                        width={30}
                        height={30}
                        viewBox="0 0 250 250"
                        style={
                            {
                                zIndex: 100000,
                                fill: 'black',
                                color: '#fff',
                                position: 'fixed',
                                top: '0px',
                                border: 1,
                                left: '0px',
                                transform: 'scale(-1.5, 1.5)'
                            }
                    }>
                        <path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path>
                        <path className={"octo-arm"}
                            d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
                            fill="currentColor"
                            style={
                                {transformOrigin: '130px 106px'}
                        }></path>
                        <path className="octo-body" d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor"></path>
                    </svg>
                </a>
            </header>
        );
    }
}
